p2p/protocol/holepunch/coordination.go

```diff
@@ -0,0 +1,302 @@
+package holepunch
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	logging "github.com/ipfs/go-log"
+	"github.com/jpillora/backoff"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/peer"
+	holepunch_pb "github.com/libp2p/go-libp2p/p2p/protocol/holepunch/pb"
```

Comment:
just `pb`, the holepunch prefix is implied by the context.

Reply:
done.